Add Title, Image, and Date to social shares

sdotson opened the pull request:
For #97 . 
<img width="675" alt="screen shot 2017-04-17 at 10 58 01 am" src="https://cloud.githubusercontent.com/assets/5664775/25093682/510c01b6-2361-11e7-811f-8cabfcbd1c76.png">
<img width="573" alt="screen shot 2017-04-17 at 10 23 54 am" src="https://cloud.githubusercontent.com/assets/5664775/25093681/51055a00-2361-11e7-9564-8f1ab7eb24fb.png">

pfarnach commented on src/components/SocialBtns/facebookSDK.js:
```diff
@@ -0,0 +1,32 @@
+ /* eslint-disable */
```
Make sure to enable eslint at the end of the file

http://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments

sdotson replied:
Hey thanks for the review. Latest re-enables eslint at the end of the file

sdotson commented:
@dorono, good eye! Since the buttons don't work with localhost, I temporarily put in an arbitrary image url (the logo on staging/heroku server) to do the screenshot. The code as it stands now in this PR should be taking the `featuredImageUrl` being used in the EventDetails component and passing it down as a prop to SocialBtns

pdw207 commented on src/components/config.js:
```diff
@@ -0,0 +1,3 @@
+export const facebookAppId = '339880856413925';
```
Maybe move to `src/config.js`

pdw207 replied:
The app id is `274135313007103`

sdotson replied:
in latest.

pdw207 commented on src/components/SocialBtns/facebookSDK.js:
```diff
@@ -0,0 +1,33 @@
+      fjs.parentNode.insertBefore(js, fjs);
+    }(document, 'script', 'facebook-jssdk'));
+  },
+  share: function({ title, eventDate, picture, eventUrl, description }) {
```
Can you use ES6 syntax share({title ...}): {}

sdotson replied:
sure can

pdw207 commented on src/components/SocialBtns/facebookSDK.js:
```diff
@@ -0,0 +1,33 @@
+ /* eslint-disable */
```
Maybe move to `src/services`? What do you think. It doesn't feel like its a component.

pdw207 commented on src/components/SocialBtns/SocialBtns.js:
```diff
@@ -1,42 +1,57 @@
+    facebookSDK.init();
+  }
+
+  render({ title, startDate, picture, description } = this.props) {
```
Nice! Why do I never do this.

pdw207 commented:
Great job! Thanks.

sdotson commented:
I've addressed all comments but noticed something with the new facebook app id: The share dialogue won't work with everybody unless we make the app public. If it's too early for that we can designate a few testers for the app and it'll work for them.

Below are screenshots of what I see when I click facebook share now and where one could change the setting to make the app public in facebook (using a screenshot of my test app as an example).

<img width="919" alt="screen shot 2017-04-20 at 9 41 02 am" src="https://cloud.githubusercontent.com/assets/5664775/25233784/003621be-25ae-11e7-9824-5f76d0e66cb5.png">
<img width="576" alt="screen shot 2017-04-20 at 9 37 50 am" src="https://cloud.githubusercontent.com/assets/5664775/25233783/0035d614-25ae-11e7-8887-4125a87f11ff.png">

pdw207 commented:
Try it now.

sdotson commented:
Looks like that error is gone now! @pdw207

pdw207 commented:
Had to whitelist the domain name. Now it works on staging.
<img width="1273" alt="screen shot 2017-04-20 at 10 28 58 am" src="https://cloud.githubusercontent.com/assets/4714249/25235790/34740526-25b4-11e7-96a5-bfa7e09ad6a2.png">
